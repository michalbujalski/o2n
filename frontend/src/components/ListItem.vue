<template>
  <v-list-item :to="{ name: 'user-details', params: { userId: item.id } }">
    <v-list-item-content>
      <v-list-item-title
        v-html="`${item.firstName} ${item.lastName}`"
      ></v-list-item-title>
      <v-list-item-subtitle class="d-flex flex-row justify-space-between">
        <span class="list-item__email"
          ><b>Email:</b><span>{{ item.email }}</span></span
        >
        <span class="flex-grow-1"
          ><b>Company: </b><span>{{ item.company }}</span></span
        >
        <span
          ><b> Age: </b><span>{{ item.age }}</span></span
        >
      </v-list-item-subtitle>
    </v-list-item-content>
  </v-list-item>
</template>
<script>
export default {
  props: {
    item: Object
  }
};
</script>
<style scoped>
.list-item__email {
  width: 350px;
}
</style>
