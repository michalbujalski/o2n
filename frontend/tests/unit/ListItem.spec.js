import ListItem from "@/components/ListItem";
import { createLocalVue, mount } from "@vue/test-utils";
import Vuetify from "vuetify";
import VueRouter from "vue-router";
import Vue from "vue";

Vue.use(Vuetify);
const localVue = createLocalVue();
localVue.use(VueRouter);

describe("ListItem component", () => {
  let vuetify;
  let router;

  beforeEach(() => {
    vuetify = new Vuetify();
    router = new VueRouter();
  });
  it("renders correctly", () => {
    const wrapper = mount(ListItem, {
      localVue,
      vuetify,
      router,
      propsData: {
        item: {
          id: "id123!@#",
          firstName: "Firstname",
          lastName: "Lastname",
          company: "some company",
          email: "test@test.com",
          age: 33,
          address: "address 12/1",
          balance: 2000
        }
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
