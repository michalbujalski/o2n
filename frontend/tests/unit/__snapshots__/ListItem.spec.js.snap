// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders correctly 1`] = `
<a
  class="v-list-item--active v-list-item v-list-item--link theme--light"
  href="#/"
  tabindex="0"
>
  <div
    class="v-list-item__content"
  >
    <div
      class="v-list-item__title"
    >
      Firstname Lastname
    </div>
     
    <div
      class="v-list-item__subtitle d-flex flex-row justify-space-between"
    >
      <span
        class="list-item__email"
      >
        <b>
          Email:
        </b>
        <span>
          test@test.com
        </span>
      </span>
       
      <span
        class="flex-grow-1"
      >
        <b>
          Company: 
        </b>
        <span>
          some company
        </span>
      </span>
       
      <span>
        <b>
           Age: 
        </b>
        <span>
          33
        </span>
      </span>
    </div>
  </div>
</a>
`;
